config/issuetracker.go

```diff
@@ -15,6 +15,14 @@ const (
 	IssueTrackerRedmine              = "REDMINE"
 )
 
+var ValidIssueTrackerAuthTypes = map[IssueTracker][]AuthType{
+	IssueTrackerGithub: { AuthTypeNone , AuthTypeAPIToken},
+	IssueTrackerGitlab: { AuthTypeNone , AuthTypeAPIToken},
+	IssueTrackerPivotal: { AuthTypeNone , AuthTypeAPIToken },
+	IssueTrackerRedmine: { AuthTypeNone , AuthTypeAPIToken },
+	IssueTrackerJira : { AuthTypeOffline },
+}
+
```

Comment:
Instead of doing the validation in the `New` functions of each issue tracker, you can add a new function here:
```go
func (it IssueTracker) IsValidAuthType(authType AuthType) bool { ... }
```

Afterwards, you can extend `validation/validation.go` and do the appropriate check there similar to the rest of the code in that file. This will confine the responsibility of validating the configuration in the `validation` package as it is intended.

Reply:
This was my original thought process but i switched around since the validate method will only run AFTER the issuetracker has been created. I wanted to not create it at all in the first place. I will do it this way then! :)

Reply:
That's true. It's a bit unfortunate I had to init the issue tracker before the validation, but we can live with this code smell for a while. If you can suggest an alternative which would not demand us to init the issuetracker before the validation, that'd be ideal. :)

Reply:
I noticed that the reason we create the IssueTracker first is to get the perspective token acquisition instructions.
We could create another map per IssueTracker->Message but i saw that some messages are generated with a unique way(using origin or something else on a non standard way) and get the values from the exposed map.

Another way would be to expose a validation.ValidateAuthType method and use it before creating the IssueTracker. The validate package will still handle the validation but there gonna be multiple calls(2 for -now).